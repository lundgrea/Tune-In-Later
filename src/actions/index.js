export const addAlbums = albums => ({
  type: "ADD_ALBUMS",
  albums
});

export const login = user => ({
  type: "LOGIN",
  user
});

export const logout = () => ({
  type: "LOGOUT"
});

export const toggleFavorite = (id) => ({
  type: "TOGGLE_FAVORITE",
  id
});

export const createUser = user => ({
  type: "CREATE_USER",
  user
});

export const handleErrors = error => ({
  type: "HANDLE_ERROR",
  error
});

export const storeFavorites = albums => ({
  type: "STORE_FAVORITES",
  albums
});

// export const deleteFavorite = albumId => ({
//   type: "DELETE_FAVORITE",
//   albumId
// });
