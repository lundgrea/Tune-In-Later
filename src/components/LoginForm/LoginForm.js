import React, { Component } from 'react';

export class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: ''
    }
  }

  loginInputs = (e) => {
    const {name, value} = e.target
    this.setState({ [name]: value })
  }


  render() {
    return (
      <section>
        <form onSubmit={this.handleSubmit}>
        <input 
          type="text"
          value={email}
          name="email"
          placeholder='example@example.com'
          onChange={this.loginInputs}
          />
        <input 
          type="password"
          value={password}
          name="password"
          placeholder='Password'
          onChange={this.loginInputs}
          />
        <button>Login</button>
      </form>
    </section>
    )
  }
}