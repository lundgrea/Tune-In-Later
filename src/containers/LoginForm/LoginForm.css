/* .login-form {
 max-height: 35vh;
 width: 20%;
} */


.login-form {
  max-height: 30vh;
  width: 20%;
  /* background-color: #F2B54a; */
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
 }

 nav {
  max-height: 30vh;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding-right: 30px;
  background-color: white;
}
