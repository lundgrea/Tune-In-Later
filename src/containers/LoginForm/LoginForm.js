import React, { Component } from 'react';
import { connect } from 'react-redux';
import { login, handleErrors } from '../../actions';
import { loginUser } from '../../apiCalls/apiCalls';
import { storeFavorites } from "../../actions";
import { getFavorites } from '../../apiCalls/apiCalls';
import { Route, Link } from "react-router-dom";
import PropTypes from 'prop-types'

export class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    }
  }

  loginInputs = e => {
    const {name, value} = e.target
    this.setState({ [name]: value })
  }

  handleSubmit = async e => {
    e.preventDefault();

    const loggedUser= await loginUser(this.state)
    if(loggedUser.error) {
      this.props.handleErrors(loggedUser.error)
    } else {
      await this.props.login(loggedUser)
      const favorites = await getFavorites(this.props.user.id)
      await this.props.storeFavorites(favorites)
      this.setState({
        email: '',
        password: ''
      })
    }
  }

  render() {
    const { email, password } =  this.state
    if(!this.props.user){
      return (
        <section>
          <form>
          <input 
            type="text"
            value={email}
            name="email"
            placeholder='example@example.com'
            onChange={this.loginInputs}
            />
          <input 
            type="input"
            value={password}
            name="password"
            placeholder='Password'
            onChange={this.loginInputs}
            />
          <Link to='/user'>
              <button onClick={this.handleSubmit}>Login</button>
          </Link>
        </form>
        <Route exact path='/user'></Route>
      </section>
      )
    } else {
      return ('')
    }
  }
}

export const mapStateToProps = store => ({
  error: store.error,
  user: store.user
});

export const mapDispatchToProps = dispatch => ({
  login: user => dispatch(login(user)),
  handleErrors: error => dispatch(handleErrors(error)),
  storeFavorites: (favorites) => dispatch(storeFavorites(favorites))
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginForm)


LoginForm.propTypes = {
  error: PropTypes.string.isRequired,
  user: PropTypes.object.isRequired,
  login: PropTypes.func.isRequired,
  handleErrors: PropTypes.func.isRequired,
  storeFavorites: PropTypes.func.isRequired
}