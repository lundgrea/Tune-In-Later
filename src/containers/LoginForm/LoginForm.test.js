import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm, mapDispatchToProps, mapStateToProps } from './LoginForm';
import { login, handleErrors, storeFavorites } from '../../actions'
describe('LoginForm', () => {
  let wrapper;
  const mockUser = {
    id: 1, name: 'aidan', email: 'aidanmckay2000@gmail.com'
  }
  const mockLogin = jest.fn()
  const mockStoreFavorites = jest.fn()
  const mockHandleError = jest.fn()
  const mockFavorites = [
    {album_name: 'mirrorland', album_id: 12345678, primary_genre_name: 'rap'}
  ]
  beforeEach(() => {
    wrapper = shallow(
      <LoginForm
        error=''
        user={mockUser}
        login={mockLogin}
        handleErrors={mockHandleError}
        storeFavorites={mockStoreFavorites}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot() 
  });
  
  it('should call login inputs on change and set state', () => {
    const mockEvent = { target: {name: 'email', value: 'aidanmckay2000@gmail.com'}}
    wrapper.find('.login-input').simulate('change', mockEvent)
    expect(wrapper.state('email')).toEqual('aidanmckay2000@gmail.com')
  });

  describe('mapStateToProps', () => {
    it('should return a string that represents error', () => {
      const mockState = {
        error: 'Please try again',
      };
      const expected = {
        error: 'Please try again',
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });

    it('should return an object with a user', () => {
      const mockState = {
        user: mockUser
      };
      const expected = {
        user: mockUser
      }

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    });
  });
});