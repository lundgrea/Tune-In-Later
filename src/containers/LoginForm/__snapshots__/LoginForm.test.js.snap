// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot 1`] = `""`;
